### PR TITLE
Handle collection classes

### PR DESCRIPTION
**Pull request by khepin**
This allows you to use a custom collection class instead of an array.

@bezhermoso @szainmehdi @mattgrande

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -246,6 +246,56 @@ Week^ {#353
...
+
+```php
+class Collector
+{
+    use JsonSerialize;
+
+    #[Json(type: Schedule::class)]
+    public Collection $schedules;
+
+    #[Json(type: Schedule::class, collection_factory_method: '::make')]
```
Comment (bezhermoso) [inline README.md:263]: Not sure I like the `::` prefix. Can it just accept a callable?

```php
#[Json(type: Schedule::class, collection_factory: Collection::make(...))]
```

Comment (khepin) [reply README.md:263]: Doesn't look like this works.
First off the syntax is only in 8.1+ and not in 8.0.
Second it doesn't seem to be valid syntax when used in an attribute.

Comment (bezhermoso) [reply README.md:263]: Ahh, that's a bummer.

Comment (bezhermoso) [reply README.md:263]: Though, I think supporting alternate parameters for static vs instance methods might be better than the special `::methodName` notation.

Comment (khepin) [reply README.md:263]: Not sure about that. That would mean you could set both and then what ... You can throw an error, but I'd rather define an API that doesn't drive you to an error as much as possible.

Comment (khepin) [reply README.md:263]: Can you dig in a bit to what's causing you pause with the `::` ?

**Review by huskicsuad — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bezhermoso) [reply README.md:263]: Can you instead check if the method is static vs non-static via reflection?

Comment (khepin) [reply README.md]: I can indeed

**Review by bezhermoso — APPROVED** [review]

**Comment by khepin** [comment]
Merging now but not putting out a release just yet, there are other things to be merged first.

**Pull request merged**